src/prefect/tasks/redis/redis_tasks.py

```diff
@@ -0,0 +1,214 @@
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+import redis
+
+
+class RedisSet(Task):
+    """
+    Task for setting a Redis key-value pair.
+
+    Args:
+        - connection (Redis Connection , optional): if not provided, connection will be established using
```

Comment:
So, unfortunately `redis` connection objects aren't `cloudpickle`able, which means we can't attach them to this task as an attribute (because `dask` will need to `cloudpickle` the task instance to send it to a worker).

Reply:
Thanks for the detailed overview on this, very helpful in clarifying the issue! I'll re-work the task code as necessary